inference-engine/src/inference_engine/include/openvino/runtime/infer_request.hpp

```diff
@@ -55,20 +56,76 @@ class INFERENCE_ENGINE_API_CLASS(InferRequest) {
      * @brief Sets input/output data to infer
      *
      * @note Memory allocation does not happen
-     * @param name Name of input or output blob.
-     * @param data Reference to input or output blob. The type of a blob must match the network input precision and
+     * @param name Name of input or output tensor.
+     * @param tensor Reference to input or output tensor. The type of a tensor must match the network input precision
+     * and size.
+     */
+    void set_tensor(const std::string& name, const Tensor& tensor);
+
+    /**
+     * @brief Sets input data to infer
+     *
+     * @note Memory allocation does not happen
+     * @param idx Index of input tensor.
+     * @param tensor Reference to input tensor. The type of a tensor must match the network input precision and
+     * size.
+     */
+    void set_input_tensor(const std::size_t idx, const Tensor& tensor);
+
+    /**
+     * @brief Sets first input data to infer
+     *
+     * @note Memory allocation does not happen
+     * @param tensor Reference to input tensor. The type of a tensor must match the network input precision and
      * size.
      */
-    void set_blob(const std::string& name, const std::shared_ptr<ie::Blob>& data);
+    void set_input_tensor(const Tensor& tensor);
+
+    /**
+     * @brief Sets output data to infer
+     *
+     * @note Memory allocation does not happen
+     * @param idx Index of output tensor.
+     * @param tensor Reference to output tensor. The type of a tensor must match the network output precision and
+     * size.
+     */
+    void set_output_tensor(const std::size_t idx, const Tensor& tensor);
+
+    /**
+     * @brief Sets first output data to infer
+     *
+     * @note Memory allocation does not happen
+     * @param tensor Reference to output tensor. The type of a tensor must match the network output precision and
+     * size.
+     */
+    void set_output_tensor(const Tensor& tensor);
 
     /**
      * @brief Gets input/output data for inference
      *
      * @note Memory allocation does not happen
-     * @param name A name of Blob to get
-     * @return A shared pointer to a Blob with a name @p name. If a blob is not found, an exception is thrown.
+     * @param name A name of tensor to get
+     * @return A Tensor with a name @p name. If a tensor is not found, an exception is thrown.
      */
-    std::shared_ptr<ie::Blob> get_blob(const std::string& name);
+    Tensor get_tensor(const std::string& name);
+
+    /**
+     * @brief Gets input data for inference
+     *
+     * @note Memory allocation does not happen
+     * @param idx An index of input tensor to get
+     * @return A Tensor with a name @p name. If a tensor is not found, an exception is thrown.
+     */
+    Tensor get_input_tensor(const std::size_t idx = 0);
```

Comment:
I think we should have 2 independent methods:
```
Tensor get_input_tensor(size_t idx);
Tensor get_input_tensor();
```
Because user can call get_tensor without argument only if network has one input/output.

Reply:
a single argument is just a partial case with the same logic. There is no difference neither from the user side nor from the implementation side between those approaches. But your suggestion just produce slightly more lines of code.

Reply:
No here we have different cases.
Because if you use `get_input_tensor` with default argument 0, means that if network has several inputs, it always returns first input.
If we have `get_input_tensor()` without argument, it throws an exception.

Reply:
done